File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:


        max_ele = max(piles)


        min_ele = 1

        while min_ele < max_ele :

            mid = (min_ele + max_ele)//2

            res = 0
            
            for p in piles:
                
                res += math.ceil(p/mid)

            if res > h :

                min_ele = mid + 1
            
            else:

                max_ele = mid 

        return max_ele

File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.java
class Solution {
    public int maxProfit(int[] prices) {
        int n=prices.length;


        int i=0,j=1;
        int result=0;
        while(j<n)
        {
            if(prices[j]>prices[i])
            {
                result=Math.max(prices[j]-prices[i],result);
            }
            else
            {
                i=j;
            }
            j++;
        }
        return result;
    }
}
